integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiClusterResource.java

```diff
@@ -0,0 +1,821 @@
+   * Deploy only CR1 before deploying resource DR. 
+   * Here we execpet a Domain Validation Error
+   *
+   *   message: 'Domain domain4 failed due to 'Domain validation error': 
+   *   Cluster resource 'domain4-cluster-2' not found in namespace 'ns-mroefg'
+   *   Update the domain resource to correct the validation error.'
+   *
+   * None of the server including admin server should start.
+   */
+  @Test
+  @DisplayName("Verify servers on missing cluster resource are picked up when the resource is available")
+  void testMissingClusterResource() {
+
+    String domainUid = "domain4"; 
+    String cluster1Name = "cluster-1"; 
+    String cluster1Res = domainUid + "-cluster-1"; 
+    String cluster2Name = "cluster-2"; 
+    String cluster2Res = domainUid + "-cluster-2"; 
+
+    String configMapName = domainUid + "-configmap"; 
+   
+    // Delete any pre-existing resources if any
+    deleteDomainResource(domainUid, domainNamespace);
+    deleteClusterCustomResourceAndVerify(cluster1Res,domainNamespace);
+    deleteClusterCustomResourceAndVerify(cluster2Res,domainNamespace);
+
+    // Create and deploy cluster resource for one cluster (cluster-1)
+    ClusterResource cluster = createClusterResource(
+        cluster1Res, cluster1Name, domainNamespace, replicaCount);
+    logger.info("Creating cluster {0} in namespace {1}",cluster1Res, domainNamespace);
+    createClusterAndVerify(cluster);
+
+    // create and deploy domain resource
+    createModelConfigMap(domainUid,configMapName);
+    DomainResource domain = createDomainResource(domainUid,
+               domainNamespace, adminSecretName,
+        TEST_IMAGES_REPO_SECRET_NAME, encryptionSecretName,
+        MII_BASIC_IMAGE_NAME + ":" + MII_BASIC_IMAGE_TAG, configMapName);
+    // set cluster references to a non-existing cluster resource
+    domain.getSpec().withCluster(new V1LocalObjectReference().name(cluster1Res));
+    domain.getSpec().withCluster(new V1LocalObjectReference().name(cluster2Res));
+    logger.info("Creating Domain Resource {0} in namespace {1} using image {2}",
+        domainUid, domainNamespace, 
+        MII_BASIC_IMAGE_NAME + ":" + MII_BASIC_IMAGE_TAG);
+
+    OffsetDateTime timestamp = now();
+    createDomainAndVerify(domain, domainNamespace);
+
+    testUntil(withLongRetryPolicy,
+        checkDomainFailedEventWithReason(opNamespace, domainNamespace, 
+        domainUid, "Domain validation error", 
+        "Warning", timestamp),
+        logger,
+        "domain event {0} to be logged in namespace {1}",
+        "Domain validation error",
+        domainNamespace);
+
+    //verify the introspector pod is not started
+    String introspectPodNameBase = getIntrospectJobName(domainUid);
+    checkPodDoesNotExist(introspectPodNameBase, domainUid, domainNamespace);
+
+    String managedServerPrefix = domainUid + "-c1-managed-server";
+    String adminPodName   = domainUid + "-admin-server";
+
+    // check no server pod is not started
```

Comment:
The comment does not match the code.

Reply:
Modified the comment to "check no server pod is started"